ocaml/xenopsd/lib/topology.ml

```diff
@@ -95,27 +95,32 @@ let seq_range a b =
   let rec next i () = if i = b then Seq.Nil else Seq.Cons (i, next (i + 1)) in
   next a
 
-(** [gen_2n n] Generates all non-empty subsets of the set of [n] nodes. *)
-let seq_gen_2n n =
+let seq_filteri p s =
+  let rec loop i s () =
```

Comment:
Wy do we need the `()` argument? The match could also use `when` instead of `if`

Reply:
`Seq` imposes `()` for constructing a new `Seq.node` because of it delays the construction of nodes as much as possible.